lager/debug/http_server.hpp

```diff
@@ -100,128 +308,158 @@ class http_debug_server
+            return create_response_(200, "text/html", "");
+        }
 
-        struct : resource_t
+        res_t pause_resource_(req_t&&)
         {
-            using resource_t::resource_t;
-            response_t render_POST(const request_t& req) override
-            {
-                this->self.context_.dispatch(typename Debugger::pause_action{});
-                return std::make_shared<httpserver::string_response>("", 200);
-            }
-        } pause_resource_ = {*this};
+            context_.dispatch(typename Debugger::pause_action{});
+            return create_response_(200, "text/html", "");
+        }
 
-        struct : resource_t
+        res_t resume_resource_(req_t&&)
         {
-            using resource_t::resource_t;
-            response_t render_POST(const request_t& req) override
-            {
-                this->self.context_.dispatch(
-                    typename Debugger::resume_action{});
-                return std::make_shared<httpserver::string_response>("", 200);
-            }
-        } resume_resource_ = {*this};
+            context_.dispatch(typename Debugger::resume_action{});
+            return create_response_(200, "text/html", "");
+        }
 
-        struct : resource_t
+        res_t gui_resource_(req_t&& req) const
         {
-            using resource_t::resource_t;
-            response_t render_GET(const request_t& req) override
-            {
-                auto req_path = req.get_path();
-                auto rel_path =
-                    req_path == "/" ? "/gui/index.html" : "/gui/" + req_path;
-                auto full_path = this->self.resources_path() + rel_path;
-                auto content_type =
-                    detail::ends_with(full_path, ".html")  ? "text/html"
-                    : detail::ends_with(full_path, ".js")  ? "text/javascript"
-                    : detail::ends_with(full_path, ".css") ? "text/css"
-                                                           /* otherwise */
-                                                           : "text/plain";
-                return std::make_shared<httpserver::file_response>(
-                    full_path, 200, content_type);
-            }
-        } gui_resource_ = {*this};
+            auto req_path = req.target().to_string();
+            auto rel_path =
+                req_path == "/" ? "/gui/index.html" : "/gui/" + req_path;
+            auto full_path = resources_path() + rel_path;
+
+            std::ifstream ifs{full_path};
+            if (!ifs.is_open())
+                return create_response_(404, "text/html", "Not found");
+
+            /*
+              The whole file content is read synchronously into string, this
+              shouldn't be a performance issue because it's used only in gui
+              startup and has no blocking effect on the application because
+              http::server has its own thread.
+            */
```

Comment:
Nitpick: I prefer `//` style comments for C++ code, particularly for comments in the code (use `/*! ... */` for Doxygen comments.)